src/Rules/ClassDependencyTreeRule.php

```diff
@@ -112,15 +113,12 @@ private function isTypeToAnalyse(ClassReflection $classReflection): bool
     private function resolveParameterTypeClass(ParameterReflection $parameterReflection): ?Class_
     {
         $parameterType = $parameterReflection->getType();
-        if (! $parameterType instanceof TypeWithClassName) {
-            return null;
-        }
-
-        $parameterClassReflection = $parameterType->getClassReflection();
-        if (! $parameterClassReflection instanceof ClassReflection) {
+        $classReflections = $parameterType->getObjectClassReflections();
+        // XXX add support for union types
```

Comment:
unions were not supported before this PR either. just making the TODO explicit for now